### PR TITLE
Improve

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/Collector/LogCollector.php [hunk 1]
```diff
@@ -33,4 +35,12 @@ private function reset(): void
     {
         $this->messages = [];
     }
+
+    #[ArrayShape(['totalLogs' => 'int|void'])]
```
Comment (samdark) [inline src/Collector/LogCollector.php:39]: I think it can't be void, only int.

File: src/Collector/EventCollector.php [hunk 2]
```diff
@@ -38,6 +43,14 @@ private function collectEvent(object $event): void
         ];
     }
 
+    #[ArrayShape(['totalEvents' => 'int|void'])]
```
Comment (samdark) [inline src/Collector/EventCollector.php:46]: I think it cannot be void, only int.

File: src/Collector/MiddlewareCollector.php [hunk 3]
```diff
@@ -51,4 +53,12 @@ private function reset(): void
         $this->beforeStack = [];
         $this->afterStack = [];
     }
+
+    #[ArrayShape(['totalMiddlewares' => 'int|void'])]
```
Comment (samdark) [inline src/Collector/MiddlewareCollector.php:57]: I think it can't be void, only int.

File: src/Collector/RouterCollector.php [hunk 4]
```diff
@@ -41,10 +43,13 @@ public function getCollected(): array
             ];
     }
 
+    #[ArrayShape(['routeMatchTime' => 'float|int', 'matchedRoute' => 'mixed'])]
```
Comment (samdark) [inline src/Collector/RouterCollector.php:46]:
```suggestion
    #[ArrayShape(['routeMatchTime' => 'float|int', 'matchedRoute' => 'string|null'])]
```

File: src/Collector/ServiceCollector.php [hunk 5]
```diff
@@ -43,6 +45,14 @@ public function collect(
         ];
     }
 
+    #[ArrayShape(['totalServices' => 'int|void'])]
```
Comment (samdark) [inline src/Collector/ServiceCollector.php:48]:
```suggestion
    #[ArrayShape(['totalServices' => 'int'])]
```

**Review by samdark — APPROVED** [review]

File: src/Collector/WebAppInfoCollector.php [hunk 6]
```diff
@@ -19,15 +21,23 @@ final class WebAppInfoCollector implements CollectorInterface, IndexCollectorInt
...
     private float $requestProcessingTimeStopped = 0;
 
+    #[ArrayShape([
+        'applicationProcessingTime' => 'float|int',
+        'applicationPreload' => 'float|int',
+        'requestProcessingTime' => 'float|int',
+        'applicationEmit' => 'float|int',
+        'memoryPeakUsage' => 'int',
+        'memoryUsage' => 'int',
+    ])]
```
Comment (vjik) [inline src/Collector/WebAppInfoCollector.php:31]: Using psalm annotations allow don't add dependency from PhpStorm.

**Pull request merged**